mountpoint-s3-fs/README.md

```diff
@@ -0,0 +1,48 @@
+# Mountpoint S3 FS (Unstable)
+
+## ⚠️ Warning: Not Production Ready ⚠️
+This library is an implementation of the core functionality of Mountpoint S3 FS. The API interface is **unstable** and subject to **significant changes**. Do not use this library in production environments.
+
+## Overview
+Mountpoint S3 is a high-performance S3-compatible file system designed for efficient cloud storage interactions. This library provides an implementation of key Mountpoint S3 functionality but should be considered as **work in progress**.
+
+## Limitations
+- **API Instability:** The API will change soon and will introduce breaking changes.
+- **No Production Support:** At the moment this library is for internal use only.
+
+## Usage (Subject to Change)
+```rust
+#[derive(Parser, Debug)]
+#[clap(
+    name = "mount-s3",
+    about = "Mountpoint for Amazon S3",
+    version = build_info::FULL_VERSION,
+    group(
+        ArgGroup::new("cache_group")
+            .multiple(true),
+    ),
+)]
+pub struct AppCliArgs {
+    #[clap(flatten)]
+    pub cli_args: mountpoint_s3_fs::cli::CliArgs,
+}
+
+fn main() -> anyhow::Result<()> {
+    let args = AppCliArgs::parse();
+    let context = mountpoint_s3_fs::cli::ContextParams {
+        full_version: build_info::FULL_VERSION.to_string(),
+    };
+    mountpoint_s3_fs::cli::main(mountpoint_s3_fs::cli::create_s3_client, args.cli_args, context)
+}
+}
+```
+
+## Roadmap
+- Extraction of configuration layer
+- Stabilization of the API
+
+## Contributing
+Please note that the library is still in heavy development. Expect breaking changes and evolving functionality.
```

Comment:
This isn't really about contributing. Drop this or link here: https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONTRIBUTING.md